react-components/src/components/RevealContainer/RevealContainer.tsx

```diff
@@ -0,0 +1,43 @@
+import { CogniteClient } from "@cognite/sdk";
+import { useEffect, useRef, ReactNode, useState, useMemo } from "react";
+import { Cognite3DViewer } from '@cognite/reveal';
+import { RevealContext } from "./RevealContext";
+import { Color } from "three";
+
+type RevealContainerProps = {
+  color?: Color;
+  sdk: CogniteClient;
+  children?: ReactNode;
+}
+
+export default function RevealContainer({ children, sdk, color }: RevealContainerProps) {
+  const [viewer, setViewer] = useState<Cognite3DViewer>();
+  const revealDomElementRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const viewer = new Cognite3DViewer({ sdk, domElement: revealDomElementRef.current! });
+    viewer.setBackgroundColor({ color, alpha: 1 })
+    setViewer(viewer);
+    return () => {
+      viewer.dispose();
+      setViewer(undefined);
+    }
+  }, []);
+
+  const mountChildren = (viewer: Cognite3DViewer | undefined) => {
+    if (viewer === undefined) return <></>;
+    return (
+      <>
+        <RevealContext.Provider value={viewer}>
+          {children}
+        </RevealContext.Provider>
+      </>
+    )
+  }
+
+  return (
+    <div ref={revealDomElementRef}>
+      {mountChildren(viewer)}
+    </div>
+  );
```

Comment:
I'm still a bit worried that it will be hard for us to define components, e.g. UI buttons outside of this div that uses Reveal.

Reply:
Add your own component under `RevealContainer` that does whatevery you need to do. And then have a click handler that manipulates props / context. Again, we might also expose a ref. for viewer if needed.